### PR TITLE
Add experimental key

### LINKED ISSUE #687: Use the advisory to leak a secret
### Context

-   What should the challenge scenario be like?
When CVEs get published, remediation sometime has failed tremendously in the sense that a leak has been fixed, but credentials not rotated.
For this, we can use https://github.com/OWASP/wrongsecrets/security/advisories/new where we then leak a secret, after which we "fix" the vuln (E.g. a vulnerability might be that someone found that the console.log in the FE is logging a seceret, and now the fix iss that we no longer forward that to the front-end, but instead still have the secret encrypted in the challenge code).

-   What should the participant learn from completing the challenge?
Always rotate your credentials when there might be a compromise
-   For what category would the challenge be? (e.g. Docker, K8s, binary)
Docker

### Did you encounter this in real life? Could you tell us more about the scenario?
Yes, a couple of times, where I was part of a security team where the devs initially followed up bug bounty findings but then ran off once the code was changed, without changing the possibly leaked credentials

### PR DESCRIPTION
**Pull request by commjoen**
### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [x] A new challenge
-   [ ] Additional documentation
-   [ ] Something else

#### Description

This provides the Github security advisory based challenge.

### Relations

Closes #687

### Checklist:

-   [x] All the contributions made are solely the work of me and my co-authors
-   [x] I tested the changes in this PR (if applicable)
-   [x] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [ ] I added UI tests to ensure my UI changes work (when change in the overall UI, not needed if just adding a challenge)
-   [x] The PR passes pre-commit hooks and automated tests

### Additional Todos:
- [x] Get this PR and https://github.com/OWASP/wrongsecrets/security/advisories/GHSA-vv4g-7gjw-fvqw textually approved
- [x] Create the tag 1.6.8RC1
- [x] Obfuscate the key in this PR
- [ ] Create a release 1.6.8
- [ ] Publish advisory https://github.com/OWASP/wrongsecrets/security/advisories/GHSA-vv4g-7gjw-fvqw

### CODE REVIEW

**Comment by commjoen** [comment]
@bendehaan can you please review this and https://github.com/OWASP/wrongsecrets/security/advisories

File: src/main/resources/explanations/challenge35.adoc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+=== Reporting on Vulnerabilities
+
+A security researcher found a Google API key and together with the project leader https://github.com/commjoen[@commjoen] made a https://github.com/OWASP/wrongsecrets/security/advisories/GHSA-vv4g-7gjw-fvqw[Security Advisory]. The only thing @commjoen dit wrong, was actually publish the API key as part of the advisory. Can you spot the key?
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge35.adoc:3]:
```suggestion
A security researcher found a Google API key and together with the project leader https://github.com/commjoen[@commjoen] made a GitHub security advisory. The only thing @commjoen did wrong was publish the API key as part of the advisory. Can you spot the key?
```

Comment (bendehaan) [reply src/main/resources/explanations/challenge35.adoc:3]: Maybe a direct link is too easy?

File: src/main/resources/explanations/challenge35_reason.adoc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+*Why we need to be careful with vulnerability reports*
+
+When you report a vulnerability, or when you publish a security advisory, always be careful with the datails you spread with them. Hardcoded secrets found, especially those harder to rotate, should not be put into your security report itself and/or the publication.
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge35_reason.adoc:3]:
```suggestion
*Why we need to be careful with vulnerability reports*

When you report a vulnerability or publish a security advisory, always be careful with the information you spread with them. Exact values of found hardcoded secrets, especially those harder to rotate, should not be put into your security report and/or the publication.
```

**Review by bendehaan — APPROVED** [review]

**Pull request merged**